develop/guide/ndarray/01-ndarray-intro.md

```diff
@@ -0,0 +1,161 @@
+# An Intro: Manipulate Data the MXNet Way with NDArray
+
+## Overview
+This guide
+will introduce you to how data is handled with MXNet. You will learn the basics
+about MXNet's multi-dimensional array format, `ndarray`.
+
+The content was
+extracted and simplified from the gluon tutorials in [The Straight
+Dope](https://gluon.mxnet.io/).
+
+## Prerequisites
+* [MXNet installed in a Python
+environment](https://mxnet.incubator.apache.org/install/index.html?language=Python).
+* Python 2.7.x or Python 3.x
+
+
+## Getting started
+
+In this chapter, we'll get
+you going with the basic functionality. Don't worry if you don't understand any
+of the basic math, like element-wise operations or normal distributions. In the
+next two chapters we'll take another pass at NDArray, teaching you both the math
+you'll need and how to realize it in code.
+
+To get started, let's import
+`mxnet`. We'll also import `ndarray` from `mxnet` for convenience. We’ll make a
+habit of setting a random seed so that you always get the same results that we
+do.
+
+```{.python .input}
+import mxnet as mx
+from mxnet import nd
+```
+
+Let's start with a very simple 1-dimensional array with a {.python .input} list.
+
+```{.python .input}
+x = nd.array([1,2,3])
+print(x)
+```
+
+Now a 2-dimensional array.
+
+```{.python .input}
+y = nd.array([[1,2,3,4], [1,2,3,4], [1,2,3,4]])
+print(y)
+```
+
+Next, let's see how to create an NDArray, without any values initialized.
+Specifically, we'll create a 2D array (also called a *matrix*) with 3 rows and 4
+columns using the `.empty` function. We'll also try out `.full` which takes an
+additional parameter for what value you want to fill in the array.
+
+```{.python .input}
+x = nd.empty((3, 3))
+print(x)
+x = nd.full((3,3), 7)
+print(x)
+```
+
+`empty` just grabs some memory and hands us back a matrix without setting the
+values of any of its entries. This means that the entries can have any form of
+values, including very big ones! Typically, we'll want our matrices initialized
+and very often we want a matrix of all zeros, so we can use the `.zeros`
+function. If you're feeling experimental, try one of the several [array creation
+functions](https://mxnet.incubator.apache.org/api/{.python
```

Comment:
.python input looks odd? is that intended?

Reply:
I'm not sure, but it isn't right... I'll have to go back and fix those...